tools/MultiProjectValidator/AnalysisRules/DependencyMismatchRule.cs

```diff
@@ -3,24 +3,30 @@
 using System.Linq;
 using Microsoft.DotNet.ProjectModel;
 using System.Text;
-using ProjectSanity.AnalysisRules.DependencyMismatch;
+using MultiProjectValidator.AnalysisRules.DependencyMismatch;
 
 namespace MultiProjectValidator.AnalysisRules
 {
     public class DependencyMismatchRule : IAnalysisRule
     {
-        public AnalysisResult Evaluate(List<ProjectContext> projectContexts)
+        public AnalysisResult Evaluate(IEnumerable<ProjectContext> projectContexts)
         {
-            var targetGroupedContexts = GroupContextsByTarget(projectContexts);
+            var filteredContexts = FilterContextList(projectContexts);
+            var targetGroupedContexts = GroupContextsByTarget(filteredContexts);
 
             var failureMessages = EvaluateProjectContextTargetGroups(targetGroupedContexts);
-            var pass = failureMessages.Count == 0;
+            var pass = failureMessages.Count() == 0;
```

Comment:
The `_messages` object in Analysis Result is intended to be general purpose (including messages which might not indicate failure). For that reason I think it's reasonable to rely on the rule to determine whether or not it passed. In this case it's an implementation detail of the rule that the presence of a message indicates failure.